rest_framework/serializers.py

```diff
@@ -700,13 +695,37 @@ def to_internal_value(self, data):
         ret = []
         errors = []
 
-        for idx, item in enumerate(data):
-            if (
-                hasattr(self, 'instance')
-                and self.instance
-                and len(self.instance) > idx
-            ):
-                self.child.instance = self.instance[idx]
+        if not self.instance and self.parent and self.parent.instance:
+            rel_mgr = get_attribute(self.parent.instance, self.source_attrs)
+            self.instance = rel_mgr.all() if rel_mgr else None
```

Comment:
I'm not sure this is the best place to do this. I'd like to hear the opinion of someone more familiar with this codebase.